app/src/main/java/com/github/factotum_sdp/factotum/ui/directory/ContactDetailsFragment.kt

```diff
@@ -23,22 +24,40 @@ class ContactDetailsFragment : Fragment() {
         val button = mainView.findViewById<Button>(R.id.button) // connect the button to the layout
         button.setOnClickListener { view ->
             view.findNavController().navigate(R.id.action_contactDetailsFragment2_to_directoryFragment)} // go back to the list of contacts when the button is clicked
-        setContactDetails(mainView, ContactsList.getItems()[arguments?.getInt("id")!!]) //links the contact details to the layout
+        val contactsViewModel =
+            ViewModelProvider(requireActivity())[ContactsViewModel::class.java] // get the contacts view model
+        setContactDetails(mainView,
+            contactsViewModel.contacts.value?.get(arguments?.getInt("id")!!)!!
+        ) //links the contact details to the layout
 
+        val updateContactButton = mainView.findViewById<Button>(R.id.button_modify_contact)
+        updateContactButton.setOnClickListener { view ->
+            val bundle = Bundle()
+            bundle.putInt("id", arguments?.getInt("id")!!)
+            view.findNavController().navigate(R.id.action_contactDetailsFragment2_to_contactCreationFragment, bundle)
+        }
+
+        val deleteContactButton = mainView.findViewById<Button>(R.id.button_delete_contact)
+        deleteContactButton.setOnClickListener { view ->
+            contactsViewModel.deleteContact(contactsViewModel.contacts.value?.get(arguments?.getInt("id")!!)!!)
```

Comment:
You could use the safe call operator ?. followed by the elvis operator ?: to provide a default value.

Reply:
True but if it replaced by a default value here, I'm guessing it's probably gonna crash anyway since it's going to try and delete a contact which it couldn't find.